### ISSUE TITLE
CCTileGrid3D uses frame buffer dims for vertex calculations

### ISSUE THREAD

**Issue by totallyevil** [cocos2d/cocos2d-x-for-xna#22]
CCTileGrid3D uses the power of 2 dimensions of the capture buffer to calculate the vertex positions of the transition vertices used in some of the transitions.

CCTransitionSplitCols is affected by this. When the transition starts, the scale of the transit-to image is scaled to the power of 2 size (512 x 1024 instead of 480 x 800). This causes a distortion of the transited scene and ends with the target scene's scale distorted.

**Comment by totallyevil** [comment]
I think this is something that I introduced in my changes to CCTexture2D to get the pixel space defined correctly in the texture.

**Comment by totallyevil** [comment]
haha:

```
    public bool initPremultipliedATextureWithImage(Texture2D texture, int POTWide, int POTHigh)
    {
```

#warning "to set pixelFormat,alpha"

```
        texture2D = texture;
        m_tContentSize.width = texture2D.Width;
        m_tContentSize.height = texture2D.Height;
```

On Android and iOS, I can't make a Texture2D with a non Power-of-two width and height.

**Comment by totallyevil** [comment]
Fixed in 0.1.3a
